packages/core/src/database/postgres/service.ts

```diff
@@ -40,56 +41,77 @@ export class PostgresDatabaseService implements BaseDatabaseService {
   private internalNamespace = "ponder";
 
   private common: Common;
-  private poolConfig: PoolConfig;
   private userNamespace: string;
   private publishSchema?: string | undefined;
 
   db: HeadlessKysely<InternalTables>;
-  indexingDb: HeadlessKysely<any>;
   syncDb: HeadlessKysely<SyncStoreTables>;
+  indexingDb: HeadlessKysely<any>;
+  readonlyDb: HeadlessKysely<any>;
 
   private schema: Schema = null!;
   private buildId: string = null!;
   private heartbeatInterval?: NodeJS.Timeout;
 
   // Only need these for metrics.
-  private adminPool: Pool;
-  private indexingPool: Pool;
+  private internalPool: Pool;
   private syncPool: Pool;
+  private indexingPool: Pool;
+  private readonlyPool: Pool;
 
   constructor({
     common,
     poolConfig,
     userNamespace,
     publishSchema,
+    isReadonly = false,
   }: {
     common: Common;
     poolConfig: PoolConfig;
     userNamespace: string;
     publishSchema?: string | undefined;
+    isReadonly?: boolean;
   }) {
     this.common = common;
-    this.poolConfig = poolConfig;
     this.userNamespace = userNamespace;
     this.publishSchema = publishSchema;
 
-    this.adminPool = createPool({
+    const internalMax = 2;
+    const equalMax = Math.floor((poolConfig.max - internalMax) / 3);
+    const [readonlyMax, indexingMax, syncMax] = isReadonly
+      ? [poolConfig.max - internalMax, 0, 0]
+      : [equalMax, equalMax, equalMax];
+
+    this.internalPool = createPool({
       ...poolConfig,
-      // 10 minutes to accommodate slow sync store migrations.
-      statement_timeout: 10 * 60 * 1000,
+      max: internalMax,
+      statement_timeout: 10 * 60 * 1000, // 10 minutes to accommodate slow sync store migrations.
     });
-    this.indexingPool = createPool(this.poolConfig);
-    this.syncPool = createPool(this.poolConfig);
+    this.syncPool = createPool({ ...poolConfig, max: readonlyMax });
+    this.indexingPool = createPool({ ...poolConfig, max: indexingMax });
+    this.readonlyPool = createPool({ ...poolConfig, max: syncMax });
```

Comment:
@jaylmiller this is the section to focus on.

Reply:
let's update this `createPool` helper to set the `application_name` for the connection. not sure if it needs to be configurable, but for example, it would really assist in debugging if it was set to RAILWAY_DEPLOY_ID (or w/e they call the env var that identifies a deploy)

Reply:
that way you could see all the connections for a given deploy by doing
```sql
select * from pg_stat_activity where application_name=<deploy id>
```

Reply:
or maybe set application name to like `<deploy_id>_indexing` to distinguish between the different pools we spin up?